src/custom-blocks/iframe/index.js

```diff
@@ -0,0 +1,238 @@
+    category: 'mojblocks',
+    attributes: {
+        iFrameURL: {
+            type: "string",
+        },
+        iFrameClassName: {
+            type: "string",
+        },
+        iFrameWidth: {
+            type: "number",
+            default: 400
+        },
+        iFrameHeight: {
+            type: "number",
+            default: 300
+        },
+        iFrameBorder: {
+            type: "boolean",
+            default: false
+        },
+        iFrameCentre: {
+            type: "boolean",
+            default: false
+        },
+        iFrameCode: {
+            type: "string",
+        },
+        iFrameButton: {
+            type: "boolean",
+            default: false
+        },
+    },
+
+    edit: props => {
+
+        const {
+            attributes: {
+                iFrameURL,
+                iFrameWidth,
+                iFrameHeight,
+                iFrameBorder,
+                iFrameCentre,
+                iFrameCode,
+                iFrameButton,
+                iFrameClassName
+            },
+            className,
+            setAttributes
+        } = props;
+
+        // Set className attribute for PHP frontend to use
+        setAttributes({ iFrameClassName: className });
+
+        const setIFrameURL = newIFrameURL => {
+            setAttributes({ iFrameURL: newIFrameURL } );
+        };
+        const setIFrameWidth = newIFrameWidth => {
+            setAttributes({ iFrameWidth: newIFrameWidth } );
+        };
+        const setIFrameHeight = newIFrameHeight => {
+            setAttributes({ iFrameHeight: newIFrameHeight } );
+        };
+        const setIFrameBorder = newIFrameBorder => {
+            setAttributes({ iFrameBorder: newIFrameBorder } );
+        };
+        const setIFrameCentre = newIFrameCentre => {
+            setAttributes({ iFrameCentre: newIFrameCentre } );
+        };
+        const setIFrameCode = newIFrameCode => {
+            setAttributes({ iFrameCode: newIFrameCode } );
+            let lowercaseCode = newIFrameCode.toLowerCase();
+            if (lowercaseCode.indexOf("<iframe") >= 0 && lowercaseCode.indexOf("src") > 0) {
+                setAttributes({iFrameButton: true});
+            } else {
+                setAttributes({iFrameButton: false});
+            }
+        };
+        const readIFrameCode = x => {
+            // This code sets the URL, Width and Height for the iFrame from the code bopied in.
+            let spacedOutIFrameCode = iFrameCode.replaceAll(">"," > ");
+            let i = 0;
+            while (spacedOutIFrameCode.indexOf(" =") >= 0 || spacedOutIFrameCode.indexOf("= ") >= 0) {
+                // Remove any spaces around equals signs
+                spacedOutIFrameCode = spacedOutIFrameCode.replaceAll(" =","=")
+                spacedOutIFrameCode = spacedOutIFrameCode.replaceAll("= ","=")
+                i++;
+                if (i > 10) break;
+            }
```

Comment:
All the cleaning up the iframe, isn't it easier/more efficient just to use a regex, like `spacedOutIFrameCode.replace(/\s*=\s*/g, '=');`